src/lib/datocms/index.ts

```diff
@@ -16,12 +17,37 @@ type DatocmsRequest = {
   variables?: { [key: string]: string };
   retryCount?: number;
 };
+
+/**
+ * Expect return value of specified Query to be available, i.e. non-nullable.
+ */
+type Assert<Query> = Required<{ [key in keyof Query]: NonNullable<Query[key]> }>;
```

Comment:
Consider for instance `PageQuery`:
```
type PageQuery = { 
  __typename?: 'Query', 
  page?: { 
    __typename: 'PageRecord', 
   id: string,
  // etc...
  } | null 
}
```

* `Required<...>` removes the  `?:` on all top level properties. 
*  `[key in keyof Query]` loops over those properties.
* `NonNullable<...>` removes the null or undefined from each of the properties.

(This is a new comment after I changed the name of this type)